src/applications/facility-locator/components/FacilityTypeDropdown.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react';
+import { kebabCase } from 'lodash/fp';
+import classNames from 'classnames';
+import Downshift from 'downshift';
+
+import { facilityTypes } from '../config';
+import { keyMap } from '../utils/helpers';
+
+const FACILITY_OPTIONS = ['all', 'health', 'benefits', 'cemetery', 'vet_center'];
+
+const facilityOptionClasses = (item, selected) => classNames(
+  'dropdown-option',
+  { selected },
+  { 'icon-option': item && item !== 'all' },
+  { [`${kebabCase(item)}-icon`]: item && item !== 'all' }
```

Comment:
Is there an advantage to having these split out into different objects rather than within one object? Also, does `selected` need to be nested within an object? I'm somewhat more familiar with the following style, but am not aware of any preferences/advantages one way or the other:
```
var btnClass = classNames('btn', this.props.className, {
  'btn-pressed': this.state.isPressed,
  'btn-over': !this.state.isPressed && this.state.isHovered
});
```

Reply:
I don't think there are really any advantages other than possibly readability in this case. I also just happened to forget to compact them into one object, but after just trying it out now, I think I prefer how it looks as it is.

`{ selected }` is in an object because it toggles on the `selected` class if the item is selected (passed in as an argument to the function).